### ISSUE TITLE
./test.sh --justrun is broken since the new Firefox codesigning requirements

### ISSUE THREAD

**Issue by strugee** [EFForg/https-everywhere#2892]
With the new extension signing requirements, `./test.sh --justrun` is completely broken. HTTPS Everywhere is disabled when the browser starts up. I patched the skeleton profile's `prefs.js`, to disable the signing requirements, but that isn't actually sufficient to enable the extension - it just turns on the "Enable" button in `about:addons`. This is problematic since (at least on OS X - not sure why it's working without PR #2891 applied, though) restarting Firefox to enable an extension registers as closing the app, so `test/firefox.sh` moves on and nukes the profile directory.

I've been at this for about 3 hours now. No matter what I do, Firefox seems to disable the extension because it's a "foreign extension" (third-party install, essentially), as printed on stdout. I suspect that Mozilla tightened some logic with this at the same time they did the signature enforcement, and didn't bother to document it because they didn't realize it would break automated workflows.

The trick to enabling an extension seems to be setting the `userDisabled` JSON key in `extensions.json` to `false`, but no matter what I do, Firefox changes this to true on startup. I have no idea what's going on, and suspect that we'll have to reach out to Mozilla for help. Bugzilla, anyone?

Protip for anyone debugging this: instead of following the advice in `test/firefox/README` and doing `diff -r`, `cd` into your profile directory and initialize a Git repository.

See also #2051

**Comment by strugee** [comment]
I should mention that I've been working with Firefox 43, a Nightly built 9/13.

**Comment by jsha** [comment]
This is a really nasty outcome of Mozilla's new policy. As of 42, there is no way to disable the signing checks. If you want to run extensions for development or testing, you need to run the Developer Edition, or at least have it installed somewhere. We should probably add a check in test.sh that errors early if it detects that the Firefox version it will run is >=42 and not Developer Edition. Also, this would be a good thing to bug Mozilla about. I think this is overzealous and will harm overall extension development, since the Developer Edition is different from the mainline version.

Note: You can also install one of the unbranded editions, which won't have the signing check, but the same argument about not being exactly the same as the mainline version applies.

**Comment by strugee** [comment]
@jsha have you done any work trying to work around this? The _only_ reason I haven't filed a bug in Bugzilla yet is because I wasn't confident that I had results that were completely reproducible/accurate. If this sounds right to you, I can go ahead and reach out to Mozilla.

**Comment by strugee** [comment]
Also, the unbranded version essentially is the same. It just doesn't have the Firefox logo, etc.

**Comment by ldgbc** [comment]
The policy is delayed by 2 version due to various (numerous) issue.

**Comment by strugee** [comment]
I hit this behavior running Nightly. So testing's impacted.

**Comment by jsha** [comment]
@strugee: I think since Nightly has the enforcement bit set, you need to have another copy of Firefox installed. I haven't done much work on the workaround yet, but one simple approach would be to allow overriding the Firefox instance in test.sh like so:

```
FIREFOX=${FIREFOX:-firefox}
...
cfx --binary $FIREFOX ...
...
$FIREFOX -no-remote -profile ...
```

**Comment by strugee** [comment]
@jsha yeah, that might work. But then we'll have this problem again when enforcement hits release. Might as well fix it now, right?

**Comment by strugee** [comment]
In theory we can have the extension installed on Nightly, sans signature enforcement, but I haven't been able to figure out how to do this in practice.

**Comment by AykutCevik** [comment]
I solved this temporarily with the FirefoxDeveloperEdition which allows running unsigned extensions. I had to change the path for Firefox in several files for this.
